src/Metadata/AssemblyMetadata.Methods.cs

```diff
@@ -158,6 +170,20 @@ private EntityHandle ResolveMethodReference(MethodInfo method)
             }
         }
 
+        public bool TryGetMethodDefinitionConstructedType(MethodInfo methodInfo, out MethodBaseDefinitionMetadata metadata) {
```

Comment:
Idem. Why introducing an alternative to `TryGetMethodDefinition`?
Idiomatic C# must be used.